### PR TITLE
The OFFSET_LATEST for auto_offset_reset was not getting honored for non-existent consumer group IDs

### PR DESCRIPTION
**Pull request by phudgins-ciena**
The OFFSET_LATEST for auto_offset_reset was not getting honored for non-existent consumer group IDs.  It was always setting it to OFFSET_EARLIEST when the consumer group was not found.  This change will check for a passed-in value to use in this case.  The default, when None is passed in, is to set it to OFFSET_EARLIEST.

### CODE REVIEW

File: afkak/test/test_consumer.py [hunk 1]
```diff
@@ -1754,6 +1754,50 @@ def test_consumer_consume_committed_no_offset_stored(self):
...
+        mockclient.send_offset_request.return_value = reqs_ds[1]
+        mockclient.send_fetch_request.return_value = reqs_ds[2]
+        consumer = Consumer(mockclient, topic, part, Mock(),
+                            consumer_group=group, auto_offset_reset=OFFSET_LATEST)
+        d = consumer.start(OFFSET_COMMITTED)
+        # Make sure request for committed offset was made
+        request = OffsetFetchRequest(topic, part)
+        mockclient.send_offset_fetch_request.assert_called_once_with(group, [request])
+        # Deliver the response. -1 offset, empty metadata
+        responses = [OffsetFetchResponse(topic, part, -1, "", KAFKA_SUCCESS)]
```
Comment (twm) [inline afkak/test/test_consumer.py:1782]: The metadata is a bytes, so ``""`` should be ``b""``. I'm guessing this error came from the test above?

File: afkak/consumer.py [hunk 2]
```diff
@@ -594,7 +594,10 @@ def _handle_offset_response(self, responses):
             # Make sure we got a valid offset back. Kafka uses -1 to indicate
             # no committed offset was retrieved
             if response.offset == OFFSET_NOT_COMMITTED:
-                self._fetch_offset = OFFSET_EARLIEST
+                if self.auto_offset_reset in [None, OFFSET_EARLIEST]:
+                    self._fetch_offset = OFFSET_EARLIEST
+                elif self.auto_offset_reset in [OFFSET_LATEST]:
```
Comment (twm) [inline afkak/consumer.py:599]: I'd prefer use of `==` rather than `in` to avoid the temporary list.

**Review by twm — APPROVED** [review]
I had two very minor bits of feedback. Otherwise this looks correct. Please address my comments and merge!

File: afkak/test/test_consumer.py (hunk 1 above)
Comment (phudgins-ciena) [reply afkak/test/test_consumer.py]: Done - see the latest commit

File: afkak/consumer.py (hunk 2 above)
Comment (phudgins-ciena) [reply afkak/consumer.py]: Done and restructured the conditional check.

**Pull request merged**